### PR TITLE
fix: correct dimension for prefix removal

### PR DESCRIPTION
**Pull request by Mr-Neutr0n**
## Bug
In `FlowMatchingModel.forward`, when removing prepended prefix tokens from the model output `v_theta` (shape `[B, T, D]`), the code slices the feature dimension (dim 2) instead of the sequence/time dimension (dim 1):

```python
v_theta = v_theta[:, :, -T:]  # slices D (feature) — incorrect
```

This corrupts the output by truncating features rather than removing the prepended conditioning tokens.

## Fix
Corrected the indexing to slice along the sequence dimension (dim 1):

```python
v_theta = v_theta[:, -T:, :]  # slices T (sequence) — correct
```

This is consistent with how the same operation is done in `LMModel.forward` (`lm.py`), where `logits` has shape `[B, K, S, card]` and the slice `logits[:, :, -S:]` correctly targets the sequence dimension at index 2 for that 4D tensor.

### CODE REVIEW

**Comment by Mr-Neutr0n** [comment]
Closing this one to tidy up my open pull requests.

It's been open around five months with no review activity, and I'd rather withdraw it than leave it sitting in your queue. Nothing needed from you, and no hard feelings at all.

If the fix is still wanted, this can be reopened, or I'm happy to redo it properly against current main. Apologies for the noise.
